Applications/ApplicationsLib/ProjectData.cpp

```diff
@@ -352,3 +355,41 @@ void ProjectData::parseNonlinearSolvers(BaseLib::ConfigTree const& config)
 		    "The nonlinear solver name is not unique");
 	}
 }
+
+static std::unique_ptr<MathLib::PiecewiseLinearInterpolation>
+createPiecewiseLinearInterpolation(BaseLib::ConfigTree const& config)
+{
+	auto const coords = config.getConfParam<std::vector<double>>("coords");
+	auto const values = config.getConfParam<std::vector<double>>("values");
+	if (coords.empty() || values.empty())
+	{
+		ERR("The given co-ordinates or values vector is empty.");
+		std::abort();
+	}
+	if (coords.size() != values.size())
+	{
+		ERR("The given co-ordinates and values vector sizes are different.");
+		std::abort();
+	}
+
+	return std::unique_ptr<MathLib::PiecewiseLinearInterpolation>{
+	    new MathLib::PiecewiseLinearInterpolation{coords, values}};
```

Comment:
The `PiecewiseLinearInterpolation` ctors need to be modified. There are three possibilities:
 - Duplicate code and replace `const&` parameters with `&&` and `std::move` in initializer list.
 - Use universal reference, _i.e._ `template <typename T> PiecewiseLI(T&& coords, T&& values) : _coords(std::forward<T>(coords)), ...`.
 - Pass-by-value and `std::move` in initializer list.
I suggest to take the third option, because the vector's move operation is cheap and it covers the two cases of passed l-values and r-values w/o code duplication and templates.

cf. to "Effective modern c++" Item 41: "Consider pass by value for copyable parameters that are cheap to move and always copied."